### ISSUE TITLE
mRest - grace note interaction

### ISSUE THREAD

**Issue by ndubo** [rism-ch/verovio#1163]
The following example (from K. 550/04) shows a mysterious space at the beginning of the measure. 
![space_w_mRest](https://user-images.githubusercontent.com/33256891/66998794-26bafe00-f0d5-11e9-91af-5166b865b52a.jpg)
There are two layers; layer 2 has an `<mRest>`, layer 1 starts with a grace note (which may be the cause for the additional space being created). There is no problem when `mRest` is replaced by `mSpace`.

Here is the code:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<mei meiversion="3.0.0" xmlns="http://www.music-encoding.org/ns/mei">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title>K.550/04</title>
    <respStmt/>
   </titleStmt>
   <pubStmt/>
  </fileDesc>
 </meiHead>
 <music>
  <body>
   <mdiv xml:id="mdiv_550004">
    <score xml:id="score_480">
     <scoreDef key.mode="minor" key.pname="g" key.sig="2f" meter.count="2" meter.sym="cut" meter.unit="2" optimize="false" xml:id="scoreDef_01-02">
      <staffGrp barthru="false" xml:id="staffGrp_01">
        <staffDef clef.line="2" clef.shape="G" key.sig="0" label.abbr="Clar." lines="5" n="3" trans.diat="-1" trans.semi="-2" xml:id="staffDef_P3_sc01">
         <label xml:id="label_P3_sc01">Clarinetti<lb xml:id="lb_P3_sc01"/> in Si♭/B</label>
         <layerDef decls="#instrVoice_4" n="1" xml:id="layerDef_Clar1_sc01"/>
         <layerDef decls="#instrVoice_5" n="2" xml:id="layerDef_Clar2_sc01"/>
        </staffDef>
      </staffGrp>
     </scoreDef>
     <section xml:id="section_564">
      <measure n="88" xml:id="m88_k550_004">
       <staff n="3" xml:id="staff_40686">
        <layer n="1" xml:id="layer_40692">
         <note accid="s" dur="16" grace="unknown" oct="5" pname="f" stem.dir="up" xml:id="note_40698"/>
         <note dur="2" oct="5" pname="g" tstamp="1" xml:id="note_40704"/>
         <note accid="s" dur="16" grace="unknown" oct="5" pname="f" stem.dir="up" xml:id="note_40710"/>
         <note dur="2" oct="5" pname="g" tstamp="2" xml:id="note_40716"/>
        </layer>
        <layer n="2" xml:id="layer_40722">
         <mRest tstamp="1" xml:id="mRest_40728"/>
        </layer>
       </staff>
       <slur endid="#note_40704" staff="3" startid="#note_40698" xml:id="slur_40890"/>
       <slur endid="#note_40716" staff="3" startid="#note_40710" xml:id="slur_40902"/>
      </measure>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```

**Comment by craigsapp** [comment]
> [grace notes] may be the cause for the additional space being created

Yes, the grace notes are causing it.  Removing the grace notes fixes the problem.

> There is no problem when `mRest` is replaced by `mSpace`.

Verovio will treat the first layer as if it were the only layer in this case. Making the mRest invisible will also have the problem.  Also note that the effect is not limited to the start of the system: other measure positions on a line are also affected (see measure 2 in the example below, for example).  When invisible whole notes are added to the second staff, the problem goes away (measures 8 and 9).  This would suggest that the grace notes are somehow affected by the centering of the mRest in the measure, and removing the extra offset given to the grace notes will fix the problem.

Tests:

<img width="539" alt="Screen Shot 2019-10-17 at 8 47 42 AM" src="https://user-images.githubusercontent.com/3487289/67025384-c919b800-f0ba-11e9-9135-5ad592a9888f.png">

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2019-10-17T08:49:58" version="2.2.0-dev-8711102-dirty">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workList>
            <work>
                <title />
            </work>
        </workList>
    </meiHead>
    <music>
        <body>
            <mdiv xml:id="mdiv-0000001119790197">
                <score xml:id="score-0000000582288163">
                    <scoreDef xml:id="scoredef-0000001487239046" midi.bpm="400">
                        <staffGrp xml:id="staffgrp-0000001040795406">
                            <staffDef xml:id="staffdef-0000001883453211" n="1" lines="5">
                                <clef xml:id="clef-L2F1" shape="G" line="2" />
                                <meterSig xml:id="metersig-L3F1" count="2" sym="cut" unit="2" />
                            </staffDef>
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-L1F1">
                        <measure xml:id="measure-L1" n="0">
                            <staff xml:id="staff-0000001668446800" n="1">
                                <layer xml:id="layer-L1F1N1" n="1">
                                    <note xml:id="note-L7F1" dur="16" oct="5" pname="f" grace="unacc" accid="s" />
                                    <note xml:id="note-L8F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                    <note xml:id="note-L9F1" dur="16" oct="5" pname="f" grace="unacc" accid.ges="s" />
                                    <note xml:id="note-L10F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                </layer>
                                <layer xml:id="layer-L8F2N2" n="2">
                                    <mRest xml:id="mrest-L8F2" ploc="b" oloc="4" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L7F1-L8F1" staff="1" startid="#note-L7F1" endid="#note-L8F1" />
                            <slur xml:id="slur-L9F1-L10F1" staff="1" startid="#note-L9F1" endid="#note-L10F1" />
                        </measure>
                        <measure xml:id="measure-L11" n="2">
                            <staff xml:id="staff-L11F1N1" n="1">
                                <layer xml:id="layer-L11F1N1" n="1">
                                    <note xml:id="note-L12F1" dur="16" oct="5" pname="f" grace="unacc" accid="s" />
                                    <note xml:id="note-L13F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                    <note xml:id="note-L14F1" dur="16" oct="5" pname="f" grace="unacc" accid.ges="s" />
                                    <note xml:id="note-L15F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                </layer>
                                <layer xml:id="layer-L11F2N2" n="2">
                                    <mRest xml:id="mrest-L13F2" ploc="b" oloc="4" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L12F1-L13F1" staff="1" startid="#note-L12F1" endid="#note-L13F1" />
                            <slur xml:id="slur-L14F1-L15F1" staff="1" startid="#note-L14F1" endid="#note-L15F1" />
                        </measure>
                        <measure xml:id="measure-L16" n="3">
                            <staff xml:id="staff-L16F1N1" n="1">
                                <layer xml:id="layer-L16F1N1" n="1">
                                    <note xml:id="note-L17F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                    <note xml:id="note-L18F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                </layer>
                                <layer xml:id="layer-L16F2N2" n="2">
                                    <mRest xml:id="mrest-L17F2" ploc="b" oloc="4" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L19" n="4">
                            <staff xml:id="staff-L19F1N1" n="1">
                                <layer xml:id="layer-L19F1N1" n="1">
                                    <note xml:id="note-L20F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                    <note xml:id="note-L21F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                </layer>
                                <layer xml:id="layer-L19F2N2" n="2">
                                    <mRest xml:id="mrest-L20F2" ploc="b" oloc="4" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L22" n="5">
                            <staff xml:id="staff-L22F1N1" n="1">
                                <layer xml:id="layer-L22F1N1" n="1">
                                    <note xml:id="note-L23F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                    <note xml:id="note-L24F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                </layer>
                                <layer xml:id="layer-L22F2N2" n="2">
                                    <mRest xml:id="mrest-L23F2" ploc="b" oloc="4" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L25" n="6">
                            <staff xml:id="staff-L25F1N1" n="1">
                                <layer xml:id="layer-L25F1N1" n="1">
                                    <note xml:id="note-L26F1" dur="16" oct="5" pname="f" grace="unacc" accid="s" />
                                    <note xml:id="note-L27F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                    <note xml:id="note-L28F1" dur="16" oct="5" pname="f" grace="unacc" accid.ges="s" />
                                    <note xml:id="note-L29F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                </layer>
                                <layer xml:id="layer-L25F2N2" n="2">
                                    <mRest xml:id="mrest-L27F2" ploc="b" oloc="4" visible="false" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L26F1-L27F1" staff="1" startid="#note-L26F1" endid="#note-L27F1" />
                            <slur xml:id="slur-L28F1-L29F1" staff="1" startid="#note-L28F1" endid="#note-L29F1" />
                        </measure>
                        <measure xml:id="measure-L30" n="7">
                            <staff xml:id="staff-L30F1N1" n="1">
                                <layer xml:id="layer-L30F1N1" n="1">
                                    <note xml:id="note-L31F1" dur="16" oct="5" pname="f" grace="unacc" accid="s" />
                                    <note xml:id="note-L32F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                    <note xml:id="note-L33F1" dur="16" oct="5" pname="f" grace="unacc" accid.ges="s" />
                                    <note xml:id="note-L34F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                </layer>
                                <layer xml:id="layer-L30F2N2" n="2">
                                    <mRest xml:id="mrest-L32F2" ploc="b" oloc="4" visible="false" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L31F1-L32F1" staff="1" startid="#note-L31F1" endid="#note-L32F1" />
                            <slur xml:id="slur-L33F1-L34F1" staff="1" startid="#note-L33F1" endid="#note-L34F1" />
                        </measure>
                        <measure xml:id="measure-L35" n="8">
                            <staff xml:id="staff-L35F1N1" n="1">
                                <layer xml:id="layer-L35F1N1" n="1">
                                    <note xml:id="note-L36F1" dur="16" oct="5" pname="f" grace="unacc" accid="s" />
                                    <note xml:id="note-L37F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                    <note xml:id="note-L38F1" dur="16" oct="5" pname="f" grace="unacc" accid.ges="s" />
                                    <note xml:id="note-L39F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                </layer>
                                <layer xml:id="layer-L35F2N2" n="2">
                                    <note xml:id="note-L37F2" dur="1" oct="4" pname="c" visible="false" accid.ges="n" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L36F1-L37F1" staff="1" startid="#note-L36F1" endid="#note-L37F1" />
                            <slur xml:id="slur-L38F1-L39F1" staff="1" startid="#note-L38F1" endid="#note-L39F1" />
                        </measure>
                        <measure xml:id="measure-L40" n="9">
                            <staff xml:id="staff-L40F1N1" n="1">
                                <layer xml:id="layer-L40F1N1" n="1">
                                    <note xml:id="note-L41F1" dur="16" oct="5" pname="f" grace="unacc" accid="s" />
                                    <note xml:id="note-L42F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                    <note xml:id="note-L43F1" dur="16" oct="5" pname="f" grace="unacc" accid.ges="s" />
                                    <note xml:id="note-L44F1" dur="2" oct="5" pname="g" stem.len="5.500000" accid.ges="n" />
                                </layer>
                                <layer xml:id="layer-L40F2N2" n="2">
                                    <note xml:id="note-L42F2" dur="1" oct="4" pname="c" visible="false" accid.ges="n" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-L41F1-L42F1" staff="1" startid="#note-L41F1" endid="#note-L42F1" />
                            <slur xml:id="slur-L43F1-L44F1" staff="1" startid="#note-L43F1" endid="#note-L44F1" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by lpugin** [comment]
Fixed with 38a1f063f6e05a94be99ec6fa0ee7d41afba480f

**Comment by craigsapp** [comment]
Looks good:

<img width="678" alt="Screen Shot 2019-10-26 at 12 03 16 PM" src="https://user-images.githubusercontent.com/3487289/67624656-59a57600-f7e8-11e9-93e5-c9084a70f0c1.png">
